ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLFlowAgentRunner.java

```diff
@@ -176,6 +177,14 @@ private String parseResponse(Object output) {
 
         } else if (output instanceof ModelTensor) {
             return toJson(((ModelTensor) output).getDataAsMap());
+        } else if (output instanceof ModelTensorOutput) {
+            List<String> modelTensorList = ((ModelTensorOutput) output)
+                .getMlModelOutputs()
+                .stream()
+                .flatMap(modelTensors -> modelTensors.getMlModelTensors().stream())
+                .map(ModelTensor::getResult)
+                .collect(Collectors.toList());
+            return toJson(modelTensorList);
```

Comment:
If verbose is set to `false`, result field will be empty and `dataAsMap` have response.

@arjunkumargiri  @ylwu-amzn  Another question for all gson usage, why we don't use the capability OpenSearch provided, using XContent like 
```java
((ModelTensorOutput) output).toXContent(JsonXContent.contentBuilder(), null)
```

Reply:
Good point. Parsing `ModelTensor` also only deserializes `dataAsMap` which would fail for verbose set to `false`: https://github.com/opensearch-project/ml-commons/blob/feature/agent_framework_dev/ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLFlowAgentRunner.java#L178. In general, response should follow similar schema irrespective of `verbose` setting.